### DISCUSSION TITLE
Refusing to allow a GitHub App to create or update workflow without `workflows` permission

### DISCUSSION THREAD

**Discussion by Undin** [community/community#27072] [Actions]
<p>Hi!<br>
I currently want to create a workflow to update another workflow scheduled (see the corresponding <a href="https://github.com/Undin/intellij-rust/blob/88db06bfafc3643a48f2ddea546328faf71c1116/.github/workflows/auto-updates.yml" rel="noopener nofollow ugc">workflow</a>).<br>
It creates a new branch, makes some changes, commits them and tries to push via <code>git push https://{actor}:{token}@github.com/{repo}.git {branch_name}</code> command.</p>
<p>But it fails with <code>! [remote rejected] nightly -&gt; nightly (refusing to allow a GitHub App to create or update workflow `.github/workflows/check.yml` without `workflows` permission)</code> error message<br>
(see <a href="https://github.com/Undin/intellij-rust/runs/2701888834?check_suite_focus=true" class="inline-onebox" rel="noopener nofollow ugc">Print actor and repo · Undin/intellij-rust@88db06b · GitHub</a>).</p>
<p>I know that workflow file modification requires <code>workflow</code> permission that’s why I pass my custom PAT with the corresponding permissions. I even check scopes of the corresponding token via <code>x-oauth-scopes:</code> header (see screenshot below) to ensure that I didn’t make a mistake here.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/8/883d7f2fc56cbd852f1df5fe04a7c67bb37d29fd.png" data-download-href="/uploads/short-url/jreGVghZlhHjSSUo2g4foKq7grb.png?dl=1" title="Screen Shot 2021-05-29 at 22.51.36" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2539310/181101577-53c74f83-81c2-4f3e-9e31-210ae240e17f.png" alt="Screen Shot 2021-05-29 at 22.51.36" data-base62-sha1="jreGVghZlhHjSSUo2g4foKq7grb" width="690" height="433" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/8/883d7f2fc56cbd852f1df5fe04a7c67bb37d29fd_2_690x433.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/8/883d7f2fc56cbd852f1df5fe04a7c67bb37d29fd_2_1035x649.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/8/883d7f2fc56cbd852f1df5fe04a7c67bb37d29fd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/8/883d7f2fc56cbd852f1df5fe04a7c67bb37d29fd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-05-29 at 22.51.36</span><span class="informations">1376×864 106 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>.
<p>But it still fails.</p>
<p>Could you help me to understand where I’m wrong, please and make it work?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>A likely reason is that the PAT isn’t actually used when pushing the changes. In the linked workflow file I can’t see any <code>git commit</code> or <code>git push</code> command. However I see that <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> is called without a <code>token</code> parameter, which means it will configure Git to use the default <code>GITHUB_TOKEN</code>.</p>

**Reply by FoliageOwO** [reply]
Thanks, this really helped me a lot! :)

**Reply by pepitoenpeligro** [reply]
Amazing! This was exactly the solution!
Thank you so much <3

**Reply by 14790897** [reply]
```          
git push "https://x-access-token:${{ secrets.PAT_TOKEN }}@github.com/${{ github.repository }}.git" HEAD:main
```
even use this, I can't push
but with 
```
        uses: actions/checkout@v4
        with:
            token: ${{ secrets.PAT_TOKEN }}
```

I success

**Comment by Undin** [comment]
<blockquote>
<p>In the linked workflow file I can’t see any <code>git commit</code> or <code>git push</code> command.</p>
</blockquote>
<p>The workflow runs python script which calls <code>git push</code></p>
<blockquote>
<p>However I see that <a href="https://github.com/actions/checkout" rel="noopener nofollow ugc"> <code>actions/checkout</code> </a> is called without a <code>token</code> parameter, which means it will configure Git to use the default <code>GITHUB_TOKEN</code> .</p>
</blockquote>
<p>Oh, it actually solves my issue. Thank you a lot!</p>
